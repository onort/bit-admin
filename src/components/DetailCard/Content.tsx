import React from "react"
import cx from "classnames"

import styles from "./DetailCard.scss"

interface Props {
  className?: string
  children: React.ReactNode
}

const Content: React.SFC<Props> = props => {
  const className = cx(styles.contentContainer, props.className)
  return <div className={className}>{props.children}</div>
}

export default Content
