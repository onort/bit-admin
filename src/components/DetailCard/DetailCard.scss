@import "../../styles/main.scss";

.container {
  width: 100%;
  padding: $spacer-md $spacer-lg;
}

.half {
  display: flex;
  flex: 1;
  flex-direction: column;
  justify-content: center;
  padding: $spacer-md $spacer-lg;

  &:first-of-type {
    margin-right: 2%;
  }
}

.title {
  font-family: $font-family-secondary;
  font-size: $font-size-xlg;
  font-weight: 700;
  border-bottom: 1px solid $light-border;
  margin-bottom: $spacer-sm;
}

.itemContainer {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.itemTitle {
  font-family: $font-family-secondary;
  font-size: $font-size-lg;
  font-weight: 700;
}

.itemContent {
  display: flex;
  align-items: center;
  color: $text-color-muted;
}

.itemLink {
  color: $blue;
}
