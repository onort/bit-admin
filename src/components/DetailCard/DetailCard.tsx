import React from "react"
import cx from "classnames"

import styles from "./DetailCard.scss"
import { Paper } from "../"

interface Props {
  children: React.ReactNode
  className?: string
  width?: "full" | "half"
}

const DetailCard: React.SFC<Props> = props => {
  const className = cx(
    styles.container,
    { [styles.half]: props.width === "half" },
    props.className
  )
  return (
    <Paper className={className} elevation={2}>
      {props.children}
    </Paper>
  )
}

DetailCard.defaultProps = {
  width: "full"
}

export default DetailCard
