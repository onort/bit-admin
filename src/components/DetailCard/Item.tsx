import React from "react"
import { MdOpenInNew as LinkIcon } from "react-icons/md"
import cx from "classnames"

import styles from "./DetailCard.scss"

interface Props {
  className?: string
  content: string | undefined
  linked?: boolean
  title?: string
  url?: string
}

const Item: React.SFC<Props> = props => {
  const className = cx(styles.itemContainer, props.className)
  const textContent = props.content ? props.content : "-"
  return (
    <div className={className}>
      {props.title && <span className={styles.itemTitle}>{props.title}</span>}
      <span className={styles.itemContent}>
        {props.linked ? (
          <a className={styles.itemLink} href={props.url} target="_blank">
            {textContent}
            <LinkIcon />
          </a>
        ) : (
          textContent
        )}
      </span>
    </div>
  )
}

Item.defaultProps = {
  linked: false
}

export default Item
