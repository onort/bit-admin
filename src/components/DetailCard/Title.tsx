import React from "react"
import cx from "classnames"

import styles from "./DetailCard.scss"

interface Props {
  className?: string
  content: string
}

const Title: React.SFC<Props> = props => {
  const className = cx(styles.title, props.className)
  return <h2 className={className}>{props.content}</h2>
}

export default Title
