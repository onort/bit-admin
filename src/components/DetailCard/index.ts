import DetailCard from "./DetailCard"
import DetailCardContent from "./Content"
import DetailCardItem from "./Item"
import DetailCardTitle from "./Title"

export default DetailCard

export { DetailCardContent, DetailCardItem, DetailCardTitle }
