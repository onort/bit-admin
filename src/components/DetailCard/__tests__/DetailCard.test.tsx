import React from "react"
import { shallow } from "enzyme"

import DetailCard, { DetailCardItem, DetailCardTitle } from "../"

const testCardTitle = "TestCardTitle"
const testCardItemTitle = "TestCardItemTitle"
const testCardItemContent = "TestCardItemContent"
const testClassName = "testClassName"
const testUrl = "http://example.com"

describe("<DetailCard />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <DetailCard>
        <DetailCardTitle content={testCardTitle} />
        <DetailCardItem
          content={testCardItemContent}
          linked={true}
          url={testUrl}
          title={testCardItemTitle}
        />
      </DetailCard>
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should be able to have custom className", () => {
    const wrapper = shallow(
      <DetailCard className={testClassName}>
        <DetailCardTitle content={testCardTitle} />
      </DetailCard>
    )
    expect(wrapper.find(`.${testClassName}`).exists()).toBe(true)
  })
})
