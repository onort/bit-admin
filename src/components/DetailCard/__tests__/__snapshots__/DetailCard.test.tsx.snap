// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailCard /> should match snapshot 1`] = `
<Paper
  className="container"
  elevation={2}
>
  <Title
    content="TestCardTitle"
  />
  <Item
    content="TestCardItemContent"
    linked={true}
    title="TestCardItemTitle"
    url="http://example.com"
  />
</Paper>
`;

exports[`<DetailCard /> should match snapshot 2`] = `"<div class=\\"container elevation2 container\\"><h2 class=\\"title\\">TestCardTitle</h2><div class=\\"itemContainer\\"><span class=\\"itemTitle\\">TestCardItemTitle</span><span class=\\"itemContent\\"><a class=\\"itemLink\\" href=\\"http://example.com\\" target=\\"_blank\\">TestCardItemContent<svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\"><path d=\\"M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z\\"></path></svg></a></span></div></div>"`;
